cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/spark/SparkAppUsingFileSet.java

```diff
@@ -105,20 +107,48 @@ public void run(SparkContext context) {
       String input = context.getRuntimeArguments().get("input");
       String output = context.getRuntimeArguments().get("output");
 
-      // read the dataset
-      JavaPairRDD<Long, String> inputData = context.readFromDataset(input, Long.class, String.class);
-
-      // create a new RDD with the same key but with a new value which is the length of the string
-      JavaPairRDD<String, Integer> stringLengths = inputData.mapToPair(
-        new PairFunction<Tuple2<Long, String>, String, Integer>() {
-          @Override
-          public Tuple2<String, Integer> call(Tuple2<Long, String> pair) throws Exception {
-            return new Tuple2<>(pair._2(), pair._2().length());
-          }
-        });
-
-      // write the character count to dataset
-      context.writeToDataset(stringLengths, output, String.class, Integer.class);
+      String inputPartitionKey = context.getRuntimeArguments().get("inputKey");
+      String outputPartitinKey = context.getRuntimeArguments().get("outputKey");
+
+      // read and write datasets with dataset arguments
+      if (inputPartitionKey != null && outputPartitinKey != null) {
+        Map<String, String> inputArgs = new HashMap<>();
+        TimePartitionedFileSetArguments.setInputStartTime(inputArgs, Long.parseLong(inputPartitionKey) - 100);
+        TimePartitionedFileSetArguments.setInputEndTime(inputArgs, Long.parseLong(inputPartitionKey) + 100);
+
+        // read the dataset with user custom dataset args
+        JavaPairRDD<Long, String> customPartitionData = context.readFromDataset(input, Long.class, String.class,
+                                                                                inputArgs);
+
+        // create a new RDD with the same key but with a new value which is the length of the string
+        JavaPairRDD<String, Integer> customPartitionStringLengths = customPartitionData.mapToPair(
+          new PairFunction<Tuple2<Long, String>, String, Integer>() {
+            @Override
+            public Tuple2<String, Integer> call(Tuple2<Long, String> pair) throws Exception {
+              return new Tuple2<>(pair._2(), pair._2().length());
+            }
+          });
+
+        // write the character count to dataset with user custom dataset args
+        Map<String, String> outputArgs = new HashMap<>();
+        TimePartitionedFileSetArguments.setOutputPartitionTime(outputArgs, Long.parseLong(outputPartitinKey));
```

Comment:
So this is for not requiring the user to provide a runtime argument with the name needed by TimePartitionedFileSet? It seems a bit redundant to me (the user provides `outputKey` instead of `output.partition.time`, which to me the latter is clearer).